src/OpenTelemetry.Api/Internal/ActivityHelperExtensions.cs

```diff
@@ -95,6 +95,24 @@ public static object GetTagValue(this Activity activity, string tagName)
             return null;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryGetTagValue(this Activity activity, string tagName, out object tagValue)
```

Comment:
We don't have to add a new method. We could use the existing method `GetTagValue`: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4637/files#diff-343f92e68a24f0d2d6ba9f91024a52f4f4b7085b5d30597dfec9303a07c14fc2R83

Reply:
How strongly do you feel about using the existing method?

I think the TryGet is cleaner and saves an unnecessary assertion (`if returnValue != null`).

Reply:
I think the min bar is:
1. don't duplicate code (e.g. if we want to add this method, at least the existing one `GetTagValue` should be updated so it'll leverage the `TryGetTagValue`).
2. keep it consistent - examine the existing usage and make proper update.

Reply:
> saves an unnecessary assertion (if returnValue != null)

Even with the new method you have to check `(if returnValue == true)`.

Reply:
> I would be in favor of reusing the existing method in this case and avoid having to introduce more code (and related tests) just to achieve a `boolean` check instead of a `null` check.

+1

Reply:
removed